File: Cpp/3330_find_the_original_typed_string_i.cpp
// https://leetcode.cn/problems/find-the-original-typed-string-i/

#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int possibleStringCount(string word) {
        int res = 1;
        word += ' ';
        int n = word.size(), start = 0;
        for (int i = 0; i < n; ++i) {
            if (word[i] != word[start]) {
                if (i - start > 1) {
                    res += i - start - 1;
                }
                start = i;
            }
        }
        return res;
    }
};

int main() {
    Solution sol;
    string word = "abbcccc";
    cout << sol.possibleStringCount(word) << '\n';
    return 0;
}

File: Python3/3330_find_the_original_typed_string_i.py

# https://leetcode.cn/problems/find-the-original-typed-string-i/

class Solution:
    def possibleStringCount(self, word: str) -> int:
        res = 1
        word += " "
        start = 0
        for i, ch in enumerate(word):
            if ch != word[start]:
                if i - start > 1:
                    res += i - start - 1
                start = i
        return res